docs/main/create-ocp-project/README.md

```diff
@@ -23,88 +24,170 @@ TARGET_NS=sonataflow-infra
+## Persistence Pre-requisites
 
-See https://www.parodos.dev/orchestrator-helm-chart/postgresql on how to install a PostgreSQL instance. Please follow the section detailing how to install using helm. In this document, a `secret` holding the credentials is created.
+If persistence is enabled, you must have a PostgreSQL instance running in the same `namespace` as the workflows. Typically this is the `sonataflow-infra` namespace.
 
+A Secret containing the PostgreSQL credentials must exist as well. See https://www.parodos.dev/orchestrator-helm-chart/postgresql on how to install a PostgreSQL instance. Please follow the section detailing how to install it using Helm - this will create the necessary secret and credentials.
 
-## Installing helm chart 
-From `charts` folder run 
-```console
-helm install workflows/create-ocp-project create-ocp-project --namespace=${TARGET_NS}
+## Installing the Helm Chart 
+
+From the `charts` folder in this repository run:
+
+```bash
+helm install create-ocp-project ./create-ocp-project/ --namespace=$TARGET_NS 
 ```
 
-### Post-installation
+## Post-Installation
+
+After the workflow is installed, you must configure environment variables for it to function.
 
-#### Environment variables
-#### Secret
+### Environment Variables
 
-We also need to set the following environment variables:
-* JIRA_API_TOKEN
-* OCP_API_SERVER_TOKEN
+Gather the following values for environment variables before moving on to the next section.
 
-To do so, edit the secret `${WORKFLOW_NAME}-creds` and set those values and the one of `NOTIFICATIONS_BEARER_TOKEN`:
+| Environment variable  | Description | Mandatory | Default value |
+|-----------------------|-------------|-----------|---------------|
+| `BACKSTAGE_NOTIFICATIONS_URL`      | The Backstage server URL for notifications, e.g `https://backstage.myhost.com` | ✅ | |
+| `NOTIFICATIONS_BEARER_TOKEN`      | The authorization bearer token to use to send notifications. This can be found in the Backstage configuration under `app.backend.auth.externalAccess` ([example](https://github.com/parodos-dev/orchestrator-helm-operator/blob/main/docs/release-1.2/existing-rhdh.md#app-config-configmap)) | ✅ | |
+| `JIRA_URL`      | The Jira server URL, e.g `https://your-instance.atlassian.net` | ✅ | |
+| `JIRA_USERNAME`      | The Jira username, e.g `you@company.com` | ✅ | |
+| `JIRA_API_TOKEN`      | The Jira API token associated with the username. If you're using Jira cloud, you can obtain a token [at this link](https://id.atlassian.com/manage-profile/security/api-tokens). | ✅ | |
+| `OCP_API_SERVER_URL`      | The OpenShift API server url, e.g `https://api.cluster.hostname.com:6443` | ✅ | |
+| `OCP_API_SERVER_TOKEN`      | The authorization bearer token to use when sending request to OpenShift | ✅ | |
+
+To obtain an OpenShift API token, create a Service Account, assign permissions to it, and request a token:
+
+```bash
+oc create sa orchestrator-ocp-api
```

Comment:
@gabriel-farache can you confirm the SA permissions? I tried `admin` and `self-provisioner` roles, but those failed. I had to use `cluster-admin` to get the project creation to work.

Reply:
It's interesting that `admin` cluster role failed, when checking why:
```
$ oc adm policy add-cluster-role-to-user admin  -z orchestrator-ocp-api3
clusterrole.rbac.authorization.k8s.io/admin added: "orchestrator-ocp-api3"

$ oc auth can-i get project --as=system:serviceaccount:rhdh-operator:orchestrator-ocp-api3
Warning: resource 'projects' is not namespace scoped in group 'project.openshift.io'

yes

$oc auth can-i create project --as=system:serviceaccount:rhdh-operator:orchestrator-ocp-api3
Warning: resource 'projects' is not namespace scoped in group 'project.openshift.io'

no
```

In the workflow we are creating then reading the project created to get its state.
To create the project we are sending a POST request to ` /apis/project.openshift.io/v1/projects`, and we can see that the `admin` cannot `create` a `project`

But
```
$ oc auth can-i create projectrequest --as=system:serviceaccount:rhdh-operator:orchestrator-ocp-api3
Warning: resource 'projectrequests' is not namespace scoped in group 'project.openshift.io'

yes
```
It appears that what we are doing is wrong, instead, we should send a POST request to ` /apis/project.openshift.io/v1/projectRequests` for which the `admin` and `self-provisioner` cluster roles are set to allow its creation.
I will update the workflow to stick to the OCP way of creating project.

FYI `self-provisioner` only is not enough:
```
$ oc adm policy add-cluster-role-to-user self-provisioner  -z orchestrator-ocp-api4
clusterrole.rbac.authorization.k8s.io/self-provisioner added: "orchestrator-ocp-api4"

$ oc auth can-i create projectrequest --as=system:serviceaccount:rhdh-operator:orchestrator-ocp-api4
Warning: resource 'projectrequests' is not namespace scoped in group 'project.openshift.io'

yes

$ oc auth can-i get project --as=system:serviceaccount:rhdh-operator:orchestrator-ocp-api4
Warning: resource 'projects' is not namespace scoped in group 'project.openshift.io'

no
```

We need the admin:
```
$ oc adm policy add-cluster-role-to-user admin  -z orchestrator-ocp-api4
clusterrole.rbac.authorization.k8s.io/admin added: "orchestrator-ocp-api4"

$ oc auth can-i get project --as=system:serviceaccount:rhdh-operator:orchestrator-ocp-api4
Warning: resource 'projects' is not namespace scoped in group 'project.openshift.io'

yes
```

Using the `cluster-admin` will work but it will be able to do anything on the cluster, whereas the `admin` may have more restricted set of permissions

Alternatively, we could add a step to create a cluster role to get the project
```
$oc create clusterrole read-project --verb=get --resource=projects
clusterrole.rbac.authorization.k8s.io/read-project created

$ oc adm policy add-cluster-role-to-user read-project  -z orchestrator-ocp-api4
clusterrole.rbac.authorization.k8s.io/read-project added: "orchestrator-ocp-api4"

$ oc auth can-i get project --as=system:serviceaccount:rhdh-operator:orchestrator-ocp-api4
Warning: resource 'projects' is not namespace scoped in group 'project.openshift.io'

yes
```

Reply:
cc @ElaiShalevRH

Reply:
> Alternatively, we could add a step to create a cluster role to get the project

You mean a cluster role that can create/list projects?

Reply:
> >Alternatively, we could add a step to create a cluster role to get the project

> You mean a cluster role that can create/list projects?

No, just `get` a project: in the workflow, once the project is created, we retrieve the project to get its status.

We do not need to `list` all of the projects as we are not checking if a project with similar name exists.
And we do not need `create` either as https://github.com/parodos-dev/serverless-workflows/pull/490 introduced a change to use `ProjectRequest` kind when creating a project instead of using a `POST` on the `Project` API.

If we want to use this alternative ( creating a cluster role), the SA will have to get assigned to `self-provisioner` (which provides the permissions to `create` a `ProjectRequest`)  and the newly create cluster role (which would provides the `get` permission on `Project`)